tests/pytest_framework/src/syslog_ng_config/renderer.py

```diff
@@ -39,16 +39,28 @@ def __render(self, re_create_config=None):
             self.__syslog_ng_config_content = ""
         if self.__syslog_ng_config["version"]:
             self.__render_version()
+        if self.__syslog_ng_config["global_options"]:
+            self.__render_global_options()
         if self.__syslog_ng_config["sources"]:
             self.__render_statements(root_statement="sources", statement_name="source")
+        if self.__syslog_ng_config["filters"]:
```

Comment:
As discussed IRL, this is not the right direction, and needs a little rethinking. The proposed architecture limits that filters must come in one block, and must happen before any embedded logpath. However, in a valid syslog-ng config, there is no such ordering requirement.
The limitation will be more visible when we will introduce rewrite rules, which would not reorderable with filters in the current architecture.